client/lib/purchases/assembler.ts

```diff
@@ -26,7 +27,7 @@ function createPurchaseObject( purchase ) {
 		error: null,
 		blogCreatedDate: purchase.blog_created_date,
 		expiryDate: purchase.expiry_date,
-		expiryStatus: camelCase( purchase.expiry_status ),
+		expiryStatus: snakeToCamelCase( purchase.expiry_status ),
```

Comment:
This replaces the lodash `camelCase()` function with our custom one in calypso to help reduce the dependency on lodash (and eventually remove it). Since the `expiry_status` variable actually contains kebab-case strings, this PR adds unit tests to prove that `snakeToCamelCase()` also handles those.